vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -192,13 +202,60 @@ type clusterStateFeeder struct {
 	memorySaveMode      bool
 	controllerFetcher   controllerfetcher.ControllerFetcher
 	recommenderName     string
+	historySource       HistorySource
+	promHistoryConfig   history.PrometheusHistoryProviderConfig
+}
+
+func (feeder *clusterStateFeeder) Init() error {
+	switch feeder.historySource {
+	case Checkpoints:
+		klog.Infof("Using checkpoints as a history provider")
+		feeder.initFromCheckpoints()
+	case Prometheus:
+		klog.Infof("Using prometheus as a history provider")
+
+		provider, promInitErr := history.NewPrometheusHistoryProvider(feeder.promHistoryConfig)
+		if promInitErr != nil {
+			klog.Errorf("Could not initialize history provider")
+			return promInitErr
+		}
+
+		historyInitErr := feeder.initFromHistoryProvider(provider)
+		if historyInitErr != nil {
+			klog.Errorf("Failed to load prometheus history")
+			return historyInitErr
+		}
+	case None:
+		klog.Infof("Running without a history provider")
+	default:
+		klog.Errorf("Wrong history provider option")
+		return fmt.Errorf("history provider option is not set. Supported values: prometheus, none, checkpoint")
+	}
+	return nil
 }
 
-func (feeder *clusterStateFeeder) InitFromHistoryProvider(historyProvider history.HistoryProvider) {
+func GetHistorySourceFromArg(historySource string) (HistorySource, error) {
+	switch historySource {
+	case "checkpoint":
+		return Checkpoints, nil
```

Comment:
I think I'd prefer those "enums" to be named more explicitly like `HistorySourceCheckpoints` or `CheckpointsHistorySource`. It's a little duplicated with the types, but when used inside the package it can be unclear that this is really just an enum value.
But more of a nit.

Reply:
That was intentional, as I personally, prefer to avoid "smurf" naming, but please let me know if this is a requested change, will be happy to make it